### PR TITLE
Add missing props and fix Request.clone()

### PR DESCRIPTION
**Pull request by AngelMunoz**
update request/response bindings and most importantly fix the clone signature for the Request interface

I've been doing some stuff with request/response within https://github.com/AngelMunoz/fable-bun (name change pending because it has more than just bun at this point) and found out there were a few things missing

Please let me know if something looks odd I tried to follow the existing code

### CODE REVIEW

File: src/Fetch.fs [hunk 1]
```diff
@@ -11,6 +11,28 @@ open Browser.Types
 
 [<AutoOpen>]
 module Types =
+    [<StringEnum; RequireQualifiedAccess>]
+    type RequestDestination =
+        | [<CompiledName("")>] Empty
```
Comment (alfonsogarciacaro) [inline src/Fetch.fs:16]: Does this work? I've never used an empty string 😅

File: src/Fetch.fs [hunk 2]
```diff
@@ -11,6 +11,28 @@ open Browser.Types
...
+        | Audio
+        | [<CompiledName("audioworklet")>] AudioWorklet
```
Comment (alfonsogarciacaro) [inline src/Fetch.fs:18]: Probably we need to have a `LowerAll` Case rule so we don't need to use CompiledName in this situations.

File: src/Fetch.fs [hunk 3]
```diff
@@ -358,6 +402,16 @@ module Types =
         | KeepAlive of bool
         | Signal of AbortSignal
 
+type Response with
+    [<Emit("Response.json($0, $1)")>]
```
Comment (alfonsogarciacaro) [inline src/Fetch.fs:406]: `status` can be optional so if it's not present Fable will generate wrong code: `Response.json(foo, )`, better to use the ellipsis so Fable skips the comma if there's only one argument: `Response.json($0...)`

**Review by alfonsogarciacaro — APPROVED** [review]
Thanks, only one change in Emit is needed, I can do it myself and release :+1:

**Comment by alfonsogarciacaro** [comment]
Published as 2.5, thanks again! 🎉

File: src/Fetch.fs (hunk 1 above)
Comment (AngelMunoz) [reply src/Fetch.fs:16]: I have no idea to be honest 🤣 I hope you would tell me if that was wrong, but the property can indeed be an empty string

File: src/Fetch.fs (hunk 3 above)
Comment (AngelMunoz) [reply src/Fetch.fs:406]: Oh nice, I was not aware of that thanks!

**Pull request merged**